tests/unit/test_advanced.py

```diff
@@ -402,6 +416,43 @@ def test_get_user_state_does_not_include_correctness(self):
 
         self.assertEqual(self.block.item_state, original_item_state)
 
+    @ddt.data(
+        (None, 10, True),
+        (0, 12, True),
+        (3, 3, False),
+    )
+    @ddt.unpack
+    def test_show_answer_validation(self, max_attempts, attempts, expect_validation_error):
```

Comment:
@arbrandes Please add a docstring to this test and `test_get_correct_state` below.